dartagnan/src/main/java/com/dat3m/dartagnan/encoding/ProgramEncoder.java

```diff
@@ -78,23 +79,14 @@ public BooleanFormula encodeFullProgram() {
 
     public BooleanFormula encodeConstants() {
         List<BooleanFormula> enc = new ArrayList<>();
-        for (NonDetInt constant : context.getTask().getProgram().getConstants()) {
-            Formula formula = context.encodeFinalExpression(constant);
-            if (formula instanceof BitvectorFormula bitvector) {
-                boolean signed = constant.isSigned();
-                var bitvectorFormulaManager = context.getFormulaManager().getBitvectorFormulaManager();
-                int bitWidth = bitvectorFormulaManager.getLength(bitvector);
-                constant.getMin().ifPresent(min -> enc.add(bitvectorFormulaManager.greaterOrEquals(bitvector, bitvectorFormulaManager.makeBitvector(bitWidth, min), signed)));
-                constant.getMax().ifPresent(max -> enc.add(bitvectorFormulaManager.lessOrEquals(bitvector, bitvectorFormulaManager.makeBitvector(bitWidth, max), signed)));
-            } else if (formula instanceof IntegerFormula integer) {
-                var integerFormulaManager = context.getFormulaManager().getIntegerFormulaManager();
-                constant.getMin().ifPresent(min -> enc.add(integerFormulaManager.greaterOrEquals(integer, integerFormulaManager.makeNumber(min))));
-                constant.getMax().ifPresent(max -> enc.add(integerFormulaManager.lessOrEquals(integer, integerFormulaManager.makeNumber(max))));
-            } else {
-                throw new UnsupportedOperationException(
-                        String.format(
-                                "Program constant bounds of type %s.",
-                                context.getFormulaManager().getFormulaType(formula)));
+        for (NonDetValue value : context.getTask().getProgram().getConstants()) {
+            Formula formula = context.encodeFinalExpression(value);
+            if (formula instanceof IntegerFormula intFormula && value.getType() instanceof IntegerType intType && !intType.isMathematical()) {
+                IntegerFormulaManager integerFormulaManager = context.getFormulaManager().getIntegerFormulaManager();
+                IntegerFormula min = integerFormulaManager.makeNumber(intType.getMinimumValue(true));
+                IntegerFormula max = integerFormulaManager.makeNumber(intType.getMaximumValue(false));
+                enc.add(integerFormulaManager.greaterOrEquals(intFormula, min));
+                enc.add(integerFormulaManager.lessOrEquals(intFormula, max));
```

Comment:
I think we normally use `imgr` for this (the long version is too verbose). There is a similar change about the Boolean manager in this PR.

Reply:
I agree that the full name is unnecessarily verbose.

I took these changes from @xeren's branch. He removed signedness from nondet values due to unifying them into a single class. For BVs this doesn't matter, but for mathematical integers it does. I guess the intention here was to provide conservative bounds for the integer that are true for both signed and unsigned integers.
That being said, I don't think it reflects the language level semantics correctly.

I don't know how to solve this issue properly though.

Reply:
Can we simply add a `signed` field to the class (even if it is useless for bools)?

Reply:
I think I found a fitting solution. I think it makes sense to add a notion of metadata to these program constants and then we could store signedness there.